checkov/sca_package/runner.py

```diff
@@ -16,55 +16,72 @@
 class Runner(BaseRunner):
     check_type = CheckType.SCA_PACKAGE
 
-    def run(
+    def __init__(self):
+        self._check_class: Optional[str] = None
+        self._code_repo_path: Optional[Path] = None
+
+    def prepare_and_scan(
         self,
         root_folder: Union[str, Path],
-        external_checks_dir: Optional[List[str]] = None,
         files: Optional[List[str]] = None,
         runner_filter: RunnerFilter = RunnerFilter(),
-        collect_skip_comments: bool = True,
-    ) -> Report:
-        report = Report(self.check_type)
+    ) -> "Optional[Sequence[Dict[str, Any]]]":
 
         if not strtobool(os.getenv("ENABLE_SCA_PACKAGE_SCAN", "False")):
-            return report
+            return None
 
         # skip complete run, if flag '--check' was used without a CVE check ID
         if runner_filter.checks and all(not check.startswith("CKV_CVE") for check in runner_filter.checks):
-            return report
+            return None
 
         if not bc_integration.bc_api_key:
             logging.info("The --bc-api-key flag needs to be set to run SCA package scanning")
-            return report
+            return None
```

Comment:
Why have we changed those to `None`?

Reply:
I moved this logic from the method "run" into inner method called "prepare_and_scan"
all those cases represents cases where the scan can't be executed, so we return empty report finally.
when it was in "run", we should return "Report" object, so this is the old behvior.
now, I use the internal method "prepare_and_scan" which do the preparation and the scan itself, but the convertion to reports still happens in "run". returning None letting the external method that scan wasn't executed, and then "run" will return empty "Report"